cdap-proto/src/main/java/co/cask/cdap/proto/id/ArtifactId.java

```diff
@@ -25,14 +25,14 @@
 /**
  * Uniquely identifies an artifact.
  */
-// TODO: handle duplication with ArtifactId
-public class NamespacedArtifactId extends ElementId implements NamespacedId, ParentedId<NamespaceId> {
+// TODO: handle duplication with ArtifactId in cdap-api
+public class ArtifactId extends EntityId implements NamespacedId, ParentedId<NamespaceId> {
```

Comment:
I think it is very inconvenient to have two classes called ``ArtifactId``. The other one is in cdap-api. Should that be called ``Artifact`` instead? Since it is in cdap-api we may have to deprecate it first though, even though it is marked as ``@Beta``. @albertshau: thoughts?